### PR TITLE
feat: LT-1622 - Restrict Roles and Permissions Visible to User

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -50,12 +50,13 @@ def get(self, request, **kwargs):
         request_user, _ = get_user(request)
         super_user = is_super_user(request_user)
         can_deactivate = not is_super_user(user)
-
+        edit = not user["user"]["role"]["id"] == SUPER_USER_ROLE_ID
```
Comment (jfaracik) [inline users/views.py]: Make this a function

File: users/views.py [hunk 2]
```diff
@@ -50,12 +50,12 @@ def get(self, request, **kwargs):
         request_user, _ = get_user(request)
         super_user = is_super_user(request_user)
         can_deactivate = not is_super_user(user)
-
         context = {
             "profile": user["user"],
             "super_user": super_user,
             "super_user_role_id": SUPER_USER_ROLE_ID,
             "can_deactivate": can_deactivate,
+            "can_edit": can_deactivate,
```
Comment (stephanieboyd) [inline users/views.py]: Is the value for "can_edit" correct?

Comment (cstobart) [reply users/views.py]: unsure, will investigate

Comment (cstobart) [reply users/views.py]: it was unused

File: users/forms.py [hunk 3]
```diff
@@ -21,17 +22,20 @@ def add_user_form(request):
     )
 
 
-def edit_user_form(request, user):
+def edit_user_form(request, user_id, cannot_edit_role: bool):
```
Comment (andytaylorkainos) [inline users/forms.py]: `can_edit_role`

File: users/forms.py [hunk 4]
```diff
@@ -21,17 +22,20 @@ def add_user_form(request):
...
     return Form(
         title=strings.USER_EDIT_TITLE,
         questions=[
-            Select(
-                name="role",
-                options=get_roles(request, request.user.organisation, True),
-                title=strings.USER_ROLE_QUESTION,
-                include_default_select=False,
+            conditional(
+                not cannot_edit_role,
```
Comment (andytaylorkainos) [inline users/forms.py]: remove the `not`

File: users/views.py [hunk 5]
```diff
@@ -74,8 +74,10 @@ class EditUser(SingleFormView):
     def init(self, request, **kwargs):
         self.object_pk = kwargs["pk"]
         user = get_organisation_user(request, str(request.user.organisation), str(self.object_pk))
-        self.form = edit_user_form(request, user)
-        self.data = user
+        super_user = is_super_user(user) and request.user.lite_api_user_id == str(kwargs["pk"])
+        cannot_edit_role = user["user"]["id"] == request.user.lite_api_user_id or super_user
```
Comment (andytaylorkainos) [inline users/views.py]: `can_edit_role = user["user"]["id"] != request.user.lite_api_user_id`

**Pull request merged**